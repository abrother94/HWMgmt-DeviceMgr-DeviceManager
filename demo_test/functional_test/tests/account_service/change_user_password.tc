#!/bin/bash

# Edgecore DeviceManager
# Copyright 2020-2021 Edgecore Networks, Inc.
#
# Licensed to the Apache Software Foundation (ASF) under one
# or more contributor license agreements. See the NOTICE file
# distributed with this work for additional information
# regarding copyright ownership. The ASF licenses this file
# to you under the Apache License, Version 2.0 (the
# "License"); you may not use this file except in compliance
# with the License. You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing,
# software distributed under the License is distributed on an
# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
# KIND, either express or implied. See the License for the
# specific language governing permissions and limitations
# under the License.

./dm attach $1:$2:120:1:false
./dm setsessionservice $1:$2:"":true:300 2>&1 > /dev/null
admin_token=`./dm logindevice $1:$2:$5:$6:false| cut -d' ' -f4`
./dm createaccount $1:$2:$admin_token:admin_1:0penBmc:Administrator
admin_1_token=`./dm logindevice $1:$2:admin_1:0penBmc:false| cut -d' ' -f4`
./dm changeuserpassword $1:$2:$admin_token:admin_1:1penBmc
./dm createaccount $1:$2:$admin_1_token:operator:0penBmc:Operator
admin_1_token=`./dm logindevice $1:$2:admin_1:1penBmc:false| cut -d' ' -f4`
./dm createaccount $1:$2:$admin_1_token:operator:0penBmc:Operator
operator_token=`./dm logindevice $1:$2:operator:0penBmc:false| cut -d' ' -f4`
./dm changeuserpassword $1:$2:$operator_token:admin_1:1penBmc
./dm createaccount $1:$2:$admin_1_token:readonly:0penBmc:ReadOnlyUser
readonly_token=`./dm logindevice $1:$2:readonly:0penBmc:false| cut -d' ' -f4`
./dm changeuserpassword $1:$2:$readonly_token:operator:1penBmc
./dm deleteaccount $1:$2:$admin_token:admin_1
./dm deleteaccount $1:$2:$admin_token:operator
./dm deleteaccount $1:$2:$admin_token:readonly
./dm detach $1:$2:$admin_token

sed -e '/^\/\//d' -e 's/ip1/'${1}'/g; s/port1/'${2}'/g; s/ip2/'${3}'/g; s/port2/'${4}'/g; s/token/'${token}'/g' "$9".expected > "$9".e
