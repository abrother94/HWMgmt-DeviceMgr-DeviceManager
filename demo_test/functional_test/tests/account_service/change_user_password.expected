// Edgecore DeviceManager
// Copyright 2020-2021 Edgecore Networks, Inc.
//
// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements. See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership. The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License. You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied. See the License for the
// specific language governing permissions and limitations
// under the License.
ip1:port1 attached
admin_1 created
ip1:port1 changed
The user authentication data does not found
operator created
The user privilege is not administrator
The user readonly privilege is invalid
Please check this user account in this device
admin_1 deleted
operator deleted
Please check this user account in this device
ip1:port1 detached
